From 1fbb36ebabe2214b30cb78b068fb44eea0d67d29 Mon Sep 17 00:00:00 2001
From: lintao <lintao@rock-chips.com>
Date: Tue, 27 May 2014 16:32:17 +0800
Subject: [PATCH] fs: partitions:         Fix new partitions size limit bug.
 Make no sense compare sector_t size     to PAGE_SIZE, and that will fail
 partition add when too small part definition in     parameter. Now, NO LIMIT
 for partition size.

---
 block/partitions/mtdpart.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/block/partitions/mtdpart.c b/block/partitions/mtdpart.c
index ac2cf4d8efb5..6f3e0044bfdb 100755
--- a/block/partitions/mtdpart.c
+++ b/block/partitions/mtdpart.c
@@ -73,7 +73,7 @@ static struct mtd_partition * newpart(char *s,
 	else
 	{
 		size = memparse(s, &s);
-		if (size < PAGE_SIZE)
+		if (size < (PAGE_SIZE)>>9)
 		{
 			printk(KERN_ERR ERRP "partition size too small (%llx)\n", size);
 			return NULL;
-- 
2.35.3

